src/plugins/intel_gpu/include/intel_gpu/runtime/utils.hpp

```diff
@@ -183,4 +186,83 @@ inline std::string to_string(const T& v) {
 /// @}
 /// @endcond
 /// @}
+
+/// @brief LRU cache which remove the least recently used data when cache is full.
+template<typename TypeK, typename TypeD>
+class LRUCache {
+public:
+    struct CacheEntry {
+        typename std::deque<TypeK>::iterator    iter;
+        TypeD                                   data;
+        size_t                                  size;
+    };
+
+private:
+    std::deque<TypeK> deque;
+    std::unordered_map<TypeK, CacheEntry> cache_entries;
+    size_t capacity;
```

Comment:
Seems a bit weird for me, because the word "capacity" itself denotes some maximum bound. 
(Sounds like max_limit..) How about change this to "data_size" or "cur_data_size" or something?

Reply:
changed to curr_data_size